packages/components/src/components/domainproperties/list/models.spec.ts

```diff
@@ -1,10 +1,51 @@
 import {ListModel} from "./models";
-
+import {DEFAULT_LIST_SETTINGS} from "../../../test/data/constants";
+import getDomainDetailsJSON from "../../../test/data/property-getDomainDetails.json";
 
 describe('ListModel', () => {
-    test("", () => {
-        const model = ListModel.create({
 
-        })
+    test("new model", () => {
+        const newModel = ListModel.create(null, DEFAULT_LIST_SETTINGS);
+
+        expect(newModel.getIn(['domain', 'name'])).toEqual(undefined);
+        expect(newModel.getIn(['domain', 'domainId'])).toEqual(null);
+        expect(newModel.getIn(['name'])).toEqual(null);
+        expect(newModel.getIn(['keyType'])).toEqual(null);
+
+        expect(newModel.isNew()).toEqual(true);
+    });
+
+    test("existing model", () => {
+        const existingModel = ListModel.create(getDomainDetailsJSON);
+
+        expect(existingModel.getIn(['domain', 'name'])).toEqual('NIMHDemographics');
+        expect(existingModel.getIn(['domain', 'domainId'])).toEqual(2280);
+        expect(existingModel.getIn(['name'])).toEqual('NIMHDemographics');
+        expect(existingModel.getIn(['keyType'])).toEqual('Integer');
+        expect(existingModel.getIn(['keyName'])).toEqual('SubjectID');
+
+        expect(existingModel.isNew()).toEqual(false);
+    });
+
+    test("isValid", () => {
+        const validModel = ListModel.create(getDomainDetailsJSON);
+        expect(ListModel.isValid(validModel)).toEqual(true);
+
+        const invalidModelNoName = validModel.set('name', "") as ListModel;
+        expect(ListModel.isValid(invalidModelNoName)).toEqual(false);
+
+        const invalidModelHasUndefinedKeyType = validModel.set('keyType', undefined) as ListModel;
+        expect(ListModel.isValid(invalidModelHasUndefinedKeyType)).toEqual(false);
+
+        const invalidModelHasException = validModel.setIn(['domain', 'domainException'], {severity: "Error"}) as ListModel;
+        expect(ListModel.isValid(invalidModelHasException)).toEqual(false);
+    });
+
+    test("getOptions", () => {
```

Comment:
This test is trying to verify the part of the getOptions() function that makes sure teh keyName is set correctly right?  I don't see any calls to getOptions in this test, did you mean to call `existingModel.getOptions()` and then compare that objects keyName against the `existingModel.get('keyName')`?

if you call getOptions() and store that as a variable in your test, you could probably add some other quick checks for values of that options object.

Reply:
Oop, yes I did. Test updated.